CSharp/core-DirectLineWebSockets/DirectLineClient/Program.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace DirectLineSampleClient
+{
+    using System;
+    using System.Configuration;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Bot.Connector.DirectLine;
+    using Newtonsoft.Json;
+    using System.Threading;
+    using System.Text;
+    using WebSocketSharp;
+
+    public class Program
+    {
+        private static string directLineSecret = ConfigurationManager.AppSettings["DirectLineSecret"];
+        private static string botId = ConfigurationManager.AppSettings["BotId"];
+        private static string fromUser = "DirectLineSampleClientUser";
+
+        public static void Main(string[] args)
+        {
+            StartBotConversation().Wait();
+        }
+
+        private static async Task StartBotConversation()
+        {
+            var directLineClient = new DirectLineClient(directLineSecret);
+            
+            var conversation = await directLineClient.Conversations.StartConversationAsync();
+
+            using(var webSocketClient = new WebSocket(conversation.StreamUrl))
+            {
+                webSocketClient.OnMessage += WebSocketClient_OnMessage;
+                webSocketClient.Connect();
+
+                while (true)
+                {
+                    string input = Console.ReadLine().Trim();
+
+                    if (input.ToLower() == "exit")
+                    {
+                        break;
+                    }
+                    else
+                    {
+                        if (input.Length > 0)
+                        {
+                            Activity userMessage = new Activity
+                            {
+                                From = new ChannelAccount(fromUser),
+                                Text = input,
+                                Type = ActivityTypes.Message
+                            };
+
+                            await directLineClient.Conversations.PostActivityAsync(conversation.ConversationId, userMessage);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void WebSocketClient_OnMessage(object sender, MessageEventArgs e)
+        {
+            // avoid null reference exception when no data received
```

Comment:
One other item I just thought of: can you add some code to the sample that exchanges the Direct Line secret for a token before using it?